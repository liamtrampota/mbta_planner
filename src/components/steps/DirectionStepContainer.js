import React from 'react';
import '../../styles/StepContainer.css';

const getDirectionsFromRoute = (selectedRoute) => {
  const directions = [];
  selectedRoute.attributes.direction_names.forEach((directionName, index) => {
    const direction = {name: directionName, destination: selectedRoute.attributes.direction_destinations[index]};
    directions.push(direction)
  })
  return directions;
}

const DirectionStepContainer = ({ selectedRoute, selectDirection, selectedDirection }) => {
  const directions = getDirectionsFromRoute(selectedRoute);
  return (
    <div>
      <h2 style={{ textAlign: 'center' }}>
        Select which direction you are travelling
      </h2>
      <DirectionStep directions={directions} selectDirection={selectDirection} selectedDirection={selectedDirection} />
    </div>
  )
}


const DirectionStep = ({ directions, selectDirection, selectedDirection }) => (
  <div className="stepContainer">
    {directions.map((direction, index) => <Direction key={direction.name} direction={direction} selectDirection={selectDirection} index={index} selectedDirection={selectedDirection} />)}
  </div>
)

const Direction = ({ direction, selectDirection, index, selectedDirection }) => (
  <button style={{ display: 'flex', flexDirection:'column', alignItems: 'center', margin: '5px', width: '400px', backgroundColor: selectedDirection && selectedDirection.directionId === index ? 'lightgreen' : ''  }} onClick={()=>selectDirection({ directionId: index})}>
    <h4 style={{marginBottom: '5px'}}>
      {direction.name}
    </h4>
    <p style={{marginTop: '5px'}}>
      {direction.destination}
    </p>
  </button>
)

export default DirectionStepContainer;
