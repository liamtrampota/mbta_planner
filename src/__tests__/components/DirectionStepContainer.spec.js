import React from 'react';
import { shallow } from 'enzyme';
import DirectionStepContainer from '../../components/steps/DirectionStepContainer';

const props = {
  selectedRoute: { attributes: { direction_names: ["South", "North"], direction_destinations: ["Ashmont", "Montclair"] }},
  selectDirection: jest.fn(),
  selectedDirection: { directionId: 0 }
}

describe('DirectionStepContainer', () => {
  test('renders DirectionStepContainer', () => {
    const wrapper = shallow(<DirectionStepContainer {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
