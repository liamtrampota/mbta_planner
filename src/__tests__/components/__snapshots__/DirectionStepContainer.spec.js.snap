// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionStepContainer renders DirectionStepContainer 1`] = `ShallowWrapper {}`;
